### ISSUE TITLE
Dropdown, Dropdown Group, Dropdown - Ensure all interactive functionality is operable with the keyboard - (2047838678)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5637]
### Module: 
 43 Dropdown, Dropdown Group, Dropdown 

### Violation: 
 Ensure all interactive functionality is operable with the keyboard 

### WCAG Reference: 
- [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html) 

### Instance ID: 
 2047838678 

### Severity: 
 8 

### Description: 
 <calcite-dropdown placement="bottom-start" scale="m" type="click" calcite-hydrated="">
      <div slot="dropdown-trigger" appearance="solid" tabindex="0">Div Trigger</div>
      <calcite-dropdown-group group-title="Div trigger" aria-label="Div trigger" role="group" selection-mode="single" calcite-hydrated="">
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Grassland</calcite-dropdown-item>
        <calcite-dropdown-item active="" aria-checked="true" role="menuitemradio" tabindex="0" calcite-hydrated="">Forest</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Trail</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Park</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Lake</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Mountain</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">River</calcite-dropdown-item>
      </calcite-dropdown-group>
    </calcite-dropdown>
 

### Note: 
 This does not respond to enter or space
 

### Media Type: 
 Keyboard Accessibility 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Might be out of scope for the component, but helpful in updating the documentation so developers ensure any use cases where a custom element is used (such as a `div`), accessibility must be carried out.

For instance, updating the code to accommodate for keyboard users:

### html:
```html
<h3>Div triggered dropdown</h3>
<calcite-dropdown>
  <div slot="dropdown-trigger" appearance="solid" tabindex="0" onkeypress="openDropdown(event)" onClick="openDropdown(event)">Div Trigger</div>
  <calcite-dropdown-group group-title="Div trigger">
    <calcite-dropdown-item>Grassland</calcite-dropdown-item>
    <calcite-dropdown-item active>Forest</calcite-dropdown-item>
    <calcite-dropdown-item>Trail</calcite-dropdown-item>
    <calcite-dropdown-item>Park</calcite-dropdown-item>
    <calcite-dropdown-item>Lake</calcite-dropdown-item>
    <calcite-dropdown-item>Mountain</calcite-dropdown-item>
    <calcite-dropdown-item>River</calcite-dropdown-item>
  </calcite-dropdown-group>
</calcite-dropdown>
```

### javascript:
```js
const dropdown = document.querySelector("calcite-dropdown");

function openDropdown(event) {
  if (
    (event.type === "keypress" && event.key === " ") ||
    (event.type === "keypress" && event.key === "Enter")
  ) {
    dropdown.toggleAttribute("open");
  }
}

```

**Comment by geospatialem** [comment]
Closing in favor of the created documentation issues to come to a resolution.
